pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -957,7 +959,40 @@ public int getMaxUserNameLength() throws SQLException {
   }
 
   public int getDefaultTransactionIsolation() throws SQLException {
-    return Connection.TRANSACTION_READ_COMMITTED;
+    if (defaultTransactionIsolation == 0) {
```

Comment:
It looks like the current PG implementation caches `default_transaction_isolation`, and it requires client to-reconnect to observe the changed value. I believe it sounds like a bug, and it is not documented: https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-DEFAULT-TRANSACTION-ISOLATION

At the same time, the javadoc for `getDefaultTransactionIsolation` says `Retrieves this database's default transaction isolation level`, so we should probably refrain from caching the isolation at the driver level unless we implement cache invalidation properly